subprojects/plugins/src/test/groovy/org/gradle/api/tasks/GroovyRuntimeTest.groovy

```diff
@@ -42,8 +42,8 @@ class GroovyRuntimeTest extends AbstractProjectBuilderSpec {
         classifier << ["", "-indy"]
     }
 
-    def "inferred Groovy3 class path uses 'groovy' jars from classpath if all required pieces are found"() {
-        def groovyVersion = "3.0.9"
+    def "inferred Groovy class path uses 'groovy' jars from classpath if all required pieces are found and match the runtime Groovy version"() {
+        def groovyVersion = GroovySystem.version
```

Comment:
This test should work for the `3.0.9`, `GroovySystem.version` and `4.0` I think, see my comment on `GroovyRuntime`.